experiments/AMIP/user_io/leaderboard/utils.jl

```diff
@@ -0,0 +1,159 @@
+function isequispaced(arr::Vector)
+    return all(diff(arr) .≈ arr[begin + 1] - arr[begin])
+end
+
+"""
+
+Note, "flat" boundary conditions are imposed for the outer edges. This should
+make sense for most data sources (that are defined all over the globe but do not
+necessarily have points at the poles). Be sure to check that it makes sense for
+your data.
+
+The resampling performed here is a 0th-order constant resampling.
+
+"""
+
+function resample(
+    data::AbstractArray,
+    src_lonlat::Tuple{<:AbstractArray, <:AbstractArray},
+    dest_lonlat::Tuple{<:AbstractArray, <:AbstractArray},
+)
+
+    # Interpolations.jl wants ranges, so we have to make ranges out of the given src_lonlat
+    #
+    # NOTE: We are assuming that lonlat are equispaced. Check this!
+    vec_to_range(v) = range(v[begin], v[end], length = length(v))
+    src_lonlat_ranges = vec_to_range.(src_lonlat)
+
+    itp = Interpolations.constant_interpolation(
+        src_lonlat_ranges,
+        data,
+        extrapolation_bc = Interpolations.Flat(),
+    )
+
+    dest_lon, dest_lat = dest_lonlat
+
+    return [itp(lon, lat) for lon in dest_lon, lat in dest_lat]
+end
+
+"""
+
+Compute the integration weights for a first-order spherical integration.
+"""
+function integration_weights(lonlat::Tuple{<:AbstractArray, <:AbstractArray})
+    lon, lat = lonlat
+
+    # We are also assuming that they are in degrees
+    abs(maximum(lon)) <= 1 && error("longitude should be in degrees")
+    abs(maximum(lat)) <= 1 && error("latitude should be in degrees")
```

Comment:
If it's in radian, abs(maximum(lon)) is likely `pi` and abs(maximum(lat)) is likely `pi / 2`, so still larger than 1.

Reply:
Indeed.